packages/components/radio/src/utils.ts

```diff
@@ -0,0 +1,12 @@
+import { isNil } from 'lodash-unified'
+import type { RadioProps } from './radio'
+import type { RadioButtonProps } from './radio-button'
+
+export const getValueOrLabel = (props: RadioProps | RadioButtonProps) => {
+  // In version 2.x, if there's no props.value, props.label will act as props.value
+  // In version 3.x, remove this computed value, use props.value instead.
+  if (!isNil(props.value)) {
+    return props.value
+  }
+  return props.label
+}
```

Comment:
**Suggestion:**

Maybe we should avoid redundant code by extracting this into `element-plus/utils` instead of repeating this for every component.

Reply:
I'd also suggest that we introduce a deprecation `warning` for this API. 

```ts
useDeprecated('value', 'using label as the value will be deprecated in the next minor/major version, please consider giving explicit values for the all the components.')

```

Reply:
Before, I think the getValueOrLabel method is a business method. But it actually the same in Checkbox and Radio. It can be extracted into a public method.

The `value` API is newly added, and the `label` API has it's own function. So useDeprecated is not a good way.

Reply:
I meant that we need to inform the users that we have introduced a new way to passing values, to encourage them from migrating the code, we need to tell them that the new API is valid, and label will no longer be used `value` in next version. Maybe you can start by introducing a new hook for API renaming or something, to reduce ambiguity, this step is necessary.

Reply:
OK, I have a idea.